transport/src/main/java/io/netty/channel/local/PreferHeapByteBufAllocator.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.local;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.CompositeByteBuf;
+
+/**
+ * Wraps another {@link ByteBufAllocator} and use heapbuffers everywhere except when a direct buffer is explicit
+ * requested.
+ */
+final class PreferHeapByteBufAllocator implements ByteBufAllocator {
```

Comment:
Any thoughts on making this a bit more generic?  Something like `BufferPreferenceAllocator` that can be initialized with a boolean of direct vs heap?